third_party/move/move-compiler-v2/tests/checking/typing/loop_body_invalid.exp

```diff
@@ -0,0 +1,37 @@
+
+Diagnostics:
+error: `integer` not compatible with `()` in constant
+  ┌─ tests/checking/typing/loop_body_invalid.move:3:14
+  │
+3 │         loop 0
+  │              ^
+
+error: expected `()` but found `bool` in expression
+  ┌─ tests/checking/typing/loop_body_invalid.move:7:14
+  │
+7 │         loop false
+  │              ^^^^^
+
+error: expected `()` but found `address` in value
```

Comment:
For this and the following loop bodies, it would be better to have the scope of the complaint be the entire argument to loop.  Not sure how you're generating these, but in a situation where there is a contextual constraint on the type of a subexpression, wouldn't it be better to complain at the context level, rather than the point where the wrong value is generated?

Reply:
Technically not easily possible at the moment. Also there are tradeoffs. Sometimes it is better for the error messages to push the requirement from the context down to the leaves, sometimes not. I think more often the former is the case, which is currently happening.

Reply:
Ok, well it looks better in latest commit with context removed.